src/main/java/com/minecolonies/coremod/colony/buildings/BuildingTownHall.java

```diff
@@ -64,11 +75,46 @@ public void onUpgradeComplete(final int newLevel)
         }
     }
 
+    /**
+     * Add a colony permission event to the colony.
+     * Reduce the list by one if bigger than a treshhold.
+     * @param event the event to add.
+     */
+    public void addPermissionEvent(final PermissionEvent event)
+    {
+        if(getBuildingLevel() >= 1 && !permissionEvents.contains(event))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)